nni/nas/oneshot/pytorch/supermodule/sampling.py

```diff
@@ -420,10 +376,16 @@ def mutate(cls, module, name, memo, mutate_kwargs):
             elif module.merge_op == 'loose_end':
                 op_candidates_lc = module.ops[-1][-1]  # type: ignore
                 assert isinstance(op_candidates_lc, LayerChoice)
-                op_factory = {  # create a factory
-                    name: lambda _, __, ___: copy.deepcopy(op_candidates_lc[name])
-                    for name in op_candidates_lc.names
-                }
+                candidates = op_candidates_lc.candidates
+                def _copy(_, __, ___, op):
+                    return copy.deepcopy(op)
+
+                if isinstance(candidates, list):
+                    op_factory = [functools.partial(_copy, op=op) for op in candidates]
```

Comment:
Is this equal to `lambda *args: copy.deepcopy(op)`?
The mental overhead is massive...

Reply:
Oh `op` is the iterator so you need to shallow copy `op` each time before constructing the lambda...

Reply:
I suppose there is a better way, but I don't think it's worth the effort.